src/lib.rs

```diff
@@ -1,3 +1,95 @@
+//! A reasonably simple entity component system (ECS),
+//! with a design based on [hecs](https://github.com/Ralith/hecs).
+//!
+//! Compared to hecs, but it drops all optimisations which require additional API surface.
+//! Furthermore, it is _not_ thread-safe, uses prepared queries only,
+//! and adds a type map used to hold resources.
+//!
+//! # Examples
+//!
+//! A simple example that creates two entities with different components,
+//! and iterates over them using queries on different combinations of components:
+//!
+//! ```
+//! # use rs_ecs::*;
+//!
+//! let mut world = World::new();
+//!
+//! let entity = world.alloc();
+//! world.insert(entity, (42_u32, true));
+//!
+//! let entity = world.alloc();
+//! world.insert(entity, (23_u32, "hello".to_string()));
+//!
+//! for number in Query::<&u32>::new().iter(&world) {
+//!     println!("{}", number);
+//! }
+//!
+//! for (_entity, number, string) in Query::<(&Entity, &u32, &String)>::new().iter(&world) {
+//!     println!("{}, {}", string, number);
+//! }
+//! ```
+//!
+//! ----
+//!
+//! A more complex example using custom components, resources, and functions as systems:
```

Comment:
I think such "work examples" a usually placed in the `examples` folder as a separate binary? Especially since this commented and can be read standalone, making it into `examples/motion.rs` or something similar should work better than browsing it via the docs?

Reply:
Not sure if we really need examples. But as I was already doubting, I will remove the long example from the docs.